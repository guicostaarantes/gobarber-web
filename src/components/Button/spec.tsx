import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './index';

describe('Button unit test', () => {
  it('should render correctly', () => {
    const { container } = render(<Button>Label</Button>);
    expect(container).toMatchSnapshot();
    // expect(container).toBe('Label');
  });

  it('should call onClick when clicked', () => {
    const fn = jest.fn();
    const { getByTestId } = render(<Button onClick={fn} />);
    const button = getByTestId('button');
    fireEvent.click(button);
    expect(fn).toBeCalledTimes(1);
    fireEvent.click(button);
    expect(fn).toBeCalledTimes(2);
  });
});
