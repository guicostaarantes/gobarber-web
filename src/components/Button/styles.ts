import styled from 'styled-components';

import { CONSTANTS } from '../../styles/constants';

export const Container = styled.button`
  background: ${CONSTANTS.primaryColor};
  height: 56px;
  border-radius: 10px;
  margin: 8px;
  border: 0;
  padding: 0 16px;
  color: ${CONSTANTS.grayColor3};
  width: 100%;
  font-weight: 500;
  margin-top: 16px;
  transition: background-color 0.2s;

  &:hover {
    background: ${CONSTANTS.primaryDarkColor};
  }
`;
