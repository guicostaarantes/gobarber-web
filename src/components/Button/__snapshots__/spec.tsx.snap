// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button unit test should render correctly 1`] = `
.c0 {
  background: #ff9000;
  height: 56px;
  border-radius: 10px;
  margin: 8px;
  border: 0;
  padding: 0 16px;
  color: #312e38;
  width: 100%;
  font-weight: 500;
  margin-top: 16px;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.c0:hover {
  background: #cc7300;
}

<div>
  <button
    class="c0"
    data-testid="button"
    type="button"
  >
    Label
  </button>
</div>
`;
